### PR TITLE
Add missing unit tests for the Validator Registration Command

### LINKED ISSUE #7873: Add missing unit tests for the `PoS Module` Validator Registration Command
### Description

Based on the defined test cases, the following tests are missing from the implemented unit tests.

  - `verify()`
      - _params.validatorRegistrationFee != VALIDATOR_REGISTRATION_FEE_
        - Expectation: `Invalid transaction. Throw error at verify step.`
      - _name is longer than MAX_LENGTH_NAME characters long_
        - Expectation: `Invalid transaction. Throw error at verify step.`
      - _name is empty_
        - Expectation: `Invalid transaction. Throw error at verify step.`
      - _There exists already an entry in the name substore with store key trs.params.name_
        - Expectation: `Invalid transaction. Throw error at verify step.`
          
  - `execute()`
    - _Transaction passes verification and registerValidatorKeys function from the Validators module does not throw_

### Motivation

These test will align with defined test cases and ensure complete coverage of the scenarios.

### PR DESCRIPTION
**Pull request by mosmartin**
### What was the problem?

This PR resolves #7873

### How was it solved?

Added missing unit tests

### How was it tested?

Added unit tests and all are passing ✅
`npm t register_validator`

### CODE REVIEW

File: framework/test/unit/modules/pos/commands/register_validator.spec.ts [hunk 1]
```diff
@@ -324,6 +328,68 @@ describe('Validator registration command', () => {
...
+					chainID,
+				})
+				.createCommandVerifyContext<ValidatorRegistrationParams>(
+					validatorRegistrationCommandParamsSchema,
+				);
+
+			const result = await validatorRegistrationCommand.verify(context);
+
+			expect(result.status).toBe(VerifyStatus.FAIL);
+			expect(result.error?.message).toInclude('Insufficient transaction fee.');
```
Comment (has5aan) [inline framework/test/unit/modules/pos/commands/register_validator.spec.ts:360]: Perhaps the message could be relevant to the error intended to be checked, something of sort, "'name' is in an unsupported format", and test setup could be changed for `invalidTransaction` fee can be increased to be at least 1_000_000_000 to land on the correct error state in concern for the test case.

Comment (mosmartin) [reply framework/test/unit/modules/pos/commands/register_validator.spec.ts:360]: For the first part about having a relevant message, this is what is returned from the implemented functionality. As far as I know, there is `Code freeze` except for tests. So am not sure if we can update the messages now. We can create a new issue to address this.

Comment (mosmartin) [reply framework/test/unit/modules/pos/commands/register_validator.spec.ts:360]: As for the second part about the invalid transaction, I don't follow. Would you kindly elaborate? Thanks.

Comment (mosmartin) [reply framework/test/unit/modules/pos/commands/register_validator.spec.ts]: I removed the check as there is no impl. Creating a bug issue for this.

**Review by has5aan — APPROVED** [review]

**Comment by shuse2** [comment]
From the issue
- params.validatorRegistrationFee != VALIDATOR_REGISTRATION_FEE
Expectation: Invalid transaction. Throw error at verify step.
- There exists already an entry in the name substore with store key trs.params.name
Expectation: Invalid transaction. Throw error at verify step.

these 2 tests are missing?

**Comment by mosmartin** [comment]
> From the issue
> 
> * params.validatorRegistrationFee != VALIDATOR_REGISTRATION_FEE
>   Expectation: Invalid transaction. Throw error at verify step.
> * There exists already an entry in the name substore with store key trs.params.name
>   Expectation: Invalid transaction. Throw error at verify step.
> 
> these 2 tests are missing?

For the first test `params.validatorRegistrationFee != VALIDATOR_REGISTRATION_FEE` it's been covered by the `should return error if transaction does not have enough balance for the registration fee` test on L302.

For the next test `There exists already an entry in the name substore with store key trs.params.name`, it's been covered by the `should return error if store key name already exists in name store` test on L256

**Comment by shuse2** [comment]
@mosmartin then i think you can comment that on issue 👍

**Review by shuse2 — APPROVED** [review]

**Pull request merged**